tap_integration_tests/tests/showcase.rs

```diff
@@ -0,0 +1,722 @@
+    #[future] two_indexers_test_servers: Result<
+        (
+            ServerHandle,
+            SocketAddr,
+            ServerHandle,
+            SocketAddr,
+            ServerHandle,
+            SocketAddr,
+        ),
+        Error,
+    >,
+    #[future] requests_1: Result<Vec<(EIP712SignedMessage<Receipt>, u64)>>,
+    #[future] requests_2: Result<Vec<(EIP712SignedMessage<Receipt>, u64)>>,
+) -> Result<()> {
+    let (
+        _server_handle_1,
+        socket_addr_1,
+        _server_handle_2,
+        socket_addr_2,
+        _gateway_handle,
+        _gateway_addr,
+    ) = two_indexers_test_servers.await?;
+
+    let indexer_1_address = "http://".to_string() + &socket_addr_1.to_string();
+    let indexer_2_address = "http://".to_string() + &socket_addr_2.to_string();
+    let client_1 = HttpClientBuilder::default().build(indexer_1_address)?;
+    let client_2 = HttpClientBuilder::default().build(indexer_2_address)?;
+    let requests_1 = requests_1.await?;
+    let requests_2 = requests_2.await?;
+
+    for ((receipt_1, id_1), (receipt_2, id_2)) in requests_1.iter().zip(requests_2) {
+        let future_1: Pin<Box<dyn Future<Output = Result<(), jsonrpsee::core::Error>> + Send>> =
+            client_1.request("request", (id_1, receipt_1));
+        let future_2: Pin<Box<dyn Future<Output = Result<(), jsonrpsee::core::Error>> + Send>> =
+            client_2.request("request", (id_2, receipt_2));
+        let (result_1, result_2) = join!(future_1, future_2);
+
+        match result_1 {
+            Ok(()) => {}
+            Err(e) => panic!("Error making receipt request: {:?}", e),
+        }
+        match result_2 {
+            Ok(()) => {}
+            Err(e) => panic!("Error making receipt request: {:?}", e),
+        }
+    }
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn test_manager_wrong_aggregator_keys(
+    #[future] single_indexer_wrong_gateway_test_server: Result<
+        (ServerHandle, SocketAddr, ServerHandle, SocketAddr),
+        Error,
+    >,
+    #[future] requests_1: Result<Vec<(EIP712SignedMessage<Receipt>, u64)>>,
+    receipt_threshold_1: u64,
+) -> Result<()> {
+    let (_server_handle, socket_addr, _gateway_handle, _gateway_addr) =
+        single_indexer_wrong_gateway_test_server.await?;
+    let indexer_1_address = "http://".to_string() + &socket_addr.to_string();
+    let client_1 = HttpClientBuilder::default().build(indexer_1_address)?;
+    let requests = requests_1.await?;
+
+    let mut counter = 1;
+    for (receipt_1, id) in requests {
+        let result = client_1.request("request", (id, receipt_1)).await;
+        if (counter % receipt_threshold_1) == 0 {
+            match result {
+                Ok(()) => panic!("Should have failed signature verification"),
+                Err(_) => {}
+            }
+        } else {
+            match result {
+                Ok(()) => {}
+                Err(e) => panic!("Error making receipt request: {:?}", e),
+            }
+        }
```

Comment:
Please add comments explaining the if/else blocks

Reply:
added comments.